libs/MeshKernel/src/CurvilinearGrid/CurvilinearGrid.cpp

```diff
@@ -117,76 +121,85 @@ void CurvilinearGrid::Delete(std::shared_ptr<Polygons> polygons, UInt polygonInd
     {
         for (UInt m = 0; m < numM; ++m)
         {
-            if (!nodeBasedMask[n][m])
+            if (!nodeBasedMask(n, m))
             {
-                m_gridNodes[n][m].x = constants::missing::doubleValue;
-                m_gridNodes[n][m].y = constants::missing::doubleValue;
+                m_gridNodes(n, m).x = constants::missing::doubleValue;
+                m_gridNodes(n, m).y = constants::missing::doubleValue;
             }
         }
     }
 }
 
 void CurvilinearGrid::SetFlatCopies()
 {
-    if (m_gridNodes.empty())
+    if (lin_alg::MatrixIsEmpty(m_gridNodes))
     {
         return;
     }
 
-    m_numM = static_cast<UInt>(m_gridNodes.size());
-    m_numN = static_cast<UInt>(m_gridNodes[0].size());
+    m_numM = static_cast<UInt>(m_gridNodes.rows());
+    m_numN = static_cast<UInt>(m_gridNodes.cols());
     const auto [nodes, edges, gridIndices] = ConvertCurvilinearToNodesAndEdges();
     m_nodes = nodes;
     m_edges = edges;
     m_gridIndices = gridIndices;
```

Comment:
is it possible to use std::tie here

std::tie(m_nodes, m_edges, m_gridIndices) = ...

Should save on a copy

Reply:
Structure binding is used all over the lib. This is an instance. Many more exist. We can do this in a separate issue (GRIDEDIT-647).